ask-sdk-runtime/ask_sdk_runtime/utils.py

```diff
@@ -37,3 +42,56 @@ def user_agent_info(sdk_version, custom_user_agent):
         return user_agent
     else:
         return user_agent + " {}".format(custom_user_agent)
+
+
+class UserAgentManager(object):
+    """Static manager for environment level SDK user agent information.
+
+    Higher level frameworks using the SDK, but not building up on skill builder,
+    can use this static class and register their user agents.
+    """
+    _components = []
+    _user_agent = ''
+
+    @staticmethod
+    def get_user_agent():
+        # type: () -> str
+        """Get user agent string containing all registered components.
+
+        :return: User agent string with all registered components
+        :rtype: str
+        """
+        return UserAgentManager._user_agent
+
+    @staticmethod
+    def register_component(component_name):
+        # type: (str) -> None
+        """Registers component to user agent string.
+
+        The component will be appended to the existing user agent string.
+        Duplicate components are ignored.
+
+        :param component_name: Name of the component to be registered to
+            the user agent string
+        :type component_name: str
+        :return: None
+        """
+        if component_name not in UserAgentManager._components:
+            UserAgentManager._components.append(component_name)
+        updated_user_agent = ''
+        for component in UserAgentManager._components:
+            if updated_user_agent != '':
+                updated_user_agent = '{} {}'.format(updated_user_agent, component)
+            else:
+                updated_user_agent = component
+        UserAgentManager._user_agent = updated_user_agent
+
+    @staticmethod
+    def clear():
+        # type: () -> None
+        """Clear components, reset user agent to empty.
+
+        :return: None
+        """
+        UserAgentManager._components = []
```

Comment:
one question just come out: what if developer call the clear function after skill is created ? Then our SDK userAgent will be empty.

Reply:
- Order shouldn't matter since as the user agent is just a string containing components.
- As for the developer clearing the user agent, you are right, then can do it after the skill creation as well. Not sure how many developers will use it that way. One way to resolve it is add the sdk user agent everytime and only append UserAgentManager components later. Do you think that is a good alternative?